pkg/filesystem/subvolume.go

```diff
@@ -259,10 +262,9 @@ func listCephFSSubvolumes(ctx context.Context, clientsets *k8sutil.Clientsets, o
 				// subvolume info returns error in case of pending clone or if it is not ready
 				// it is suggested to delete the pvc before deleting the subvolume.
 				if err != nil {
-					if errors.Is(err, syscall.EAGAIN) {
-						logging.Warning("Found pending clone: %q", sv.Name)
-						logging.Warning("Please delete the pending pv if any before deleting the subvolume %s", sv.Name)
-						logging.Warning("To avoid stale resources, please scale down the cephfs deployment before deleting the subvolume.")
+					if isSubvolumeNotReady(err) || errors.Is(err, syscall.EAGAIN) {
```

Comment:
Is there a reason not to move the `errors.Is(err, syscall.EAGAIN)` into the `isSubvolumeNotReady()` function? It seems like it still makes sense as part of the not-ready detection.

Reply:
done